src/Platform/PlatformUnix.cpp

```diff
@@ -16,6 +16,7 @@
  * You should have received a copy of the GNU General Public License
  * along with Falltergeist.  If not, see <http://www.gnu.org/licenses/>.
  */
+#if defined(__unix__) || defined(__APPLE__) || defined(__linux__)
```

Comment:
The recommended approach (by the CMake folks, I mean) is to avoid globbing, but one less troublesome option could be to avoid to use it recursively. You glob src, you glob the folders where everything should be built unconditionally, but you don't glob over Platform, where conditions must be met for the files to be built.